python-package/lightgbm/dask.py

```diff
@@ -548,7 +548,9 @@ def _train(
     #   * 'num_threads': overridden to match nthreads on each Dask process
     for param_alias in _ConfigAliases.get("num_machines", "num_threads"):
         if param_alias in params:
-            _log_warning(f"Parameter {param_alias} will be ignored.")
+            # Only warn if user-set parameter will be overwritten
+            if params[param_alias] not in (-1, None):
+                _log_warning(f"Parameter {param_alias} will be ignored.")
```

Comment:
```suggestion
            # Only warn if user-set parameter will be overwritten
            val = params[param_alias]
            if val not in (-1, None):
                _log_warning(f"Parameter {param_alias}={val} will be ignored. This is automatically adjusted to match the Dask cluster. To suppress this warning, pass {param_alias}=-1 or remove {param_alias} entirely.")
```

While we're touching this, would you please consider also improving the warning? It should:

1. show what value was provided (especially now that that value determines whether or not the warning is emitted
2. describe how to avoid the warning